python_modules/dagster/dagster/_core/definitions/time_window_partition_mapping.py

```diff
@@ -155,10 +174,29 @@ def get_downstream_partitions_for_partitions(
         Filters for partitions that exist at the given current_time, fetching the current time
         if not provided.
         """
+        if isinstance(upstream_partitions_subset, AllPartitionsSubset):
+            from_partitions_subset = TimeWindowPartitionsSubset.from_all_partitions_subset(
+                upstream_partitions_subset
+            )
+        else:
+            from_partitions_subset = check.inst_param(
+                cast(BaseTimeWindowPartitionsSubset, upstream_partitions_subset),
+                "upstream_partitions_subset",
+                BaseTimeWindowPartitionsSubset,
+            )
+
         return self._map_partitions(
-            upstream_partitions_def,
-            downstream_partitions_def,
-            upstream_partitions_subset,
+            from_partitions_def=check.inst_param(
+                cast(TimeWindowPartitionsDefinition, upstream_partitions_def),
+                "upstream_partitions_def",
+                TimeWindowPartitionsDefinition,
+            ),
+            to_partitions_def=check.inst_param(
+                cast(TimeWindowPartitionsDefinition, downstream_partitions_def),
+                "downstream_partitions_def",
+                TimeWindowPartitionsDefinition,
+            ),
+            from_partitions_subset=from_partitions_subset,
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines 136 to 199]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->this seems a little copy-paste-y. anyway to consolidate?<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22058?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->